packages/eslint-plugin/src/rules/call-super-on-override.ts

```diff
@@ -0,0 +1,119 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import * as utils from '../util';
+
+type MessageIds = 'missingSuperMethodCall';
+
+export default utils.createRule<[], MessageIds>({
+  name: 'call-super-on-override',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Require overridden methods to call super.method in their body',
+      recommended: false,
+      requiresTypeChecking: false,
+    },
+    messages: {
+      missingSuperMethodCall:
+        "Use 'super{{property}}{{parameterTuple}}' to avoid missing super class method implementations",
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          topLevel: {
+            type: 'boolean',
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      'MethodDefinition[override=true][kind="method"]'(
+        node: TSESTree.MethodDefinition,
+      ): void {
+        let methodName = '',
+          methodNameIsLiteral = false,
+          methodNameIsNull = false; // don't add quotes for error message on [null] case
+
+        if (node.key.type === AST_NODE_TYPES.Identifier) {
+          methodName = node.key.name;
+        } else {
+          methodNameIsLiteral = true;
```

Comment:
[Refactor] Aha, a duplication of information! `methodNameIsLiteral` will always be `true` if `methodNameIsNull` is `true`. So this is using two variables to store a single piece of info. I'd suggest using either a union type or enum to consolidate down to one variable.

For example, the enum form might look something like this (but maybe with different names, I haven't thought too hard about these):

```ts
type MethodNameVariant = 'identifier' | 'non-null-literal' | 'null-literal';
```